homeassistant/components/reolink/number.py

```diff
@@ -62,10 +63,115 @@ class ReolinkNumberEntityDescription(
         native_step=1,
         get_min_value=lambda api, ch: api.zoom_range(ch)["focus"]["pos"]["min"],
         get_max_value=lambda api, ch: api.zoom_range(ch)["focus"]["pos"]["max"],
-        supported=lambda api, ch: api.zoom_supported(ch),
+        supported=lambda api, ch: api.supported(ch, "zoom"),
         value=lambda api, ch: api.get_focus(ch),
         method=lambda api, ch, value: api.set_zoom(ch, int(value)),
     ),
+    ReolinkNumberEntityDescription(
+        key="floodlight_brightness",
+        name="Floodlight turn on brightness",
+        icon="mdi:spotlight-beam",
+        entity_category=EntityCategory.CONFIG,
+        native_step=1,
+        native_min_value=1,
+        native_max_value=100,
+        supported=lambda api, ch: api.supported(ch, "floodLight"),
+        value=lambda api, ch: api.whiteled_brightness(ch),
+        method=lambda api, ch, value: api.set_whiteled(ch, brightness=int(value)),
+    ),
```

Comment:
What's the difference between this number entity and the light entity for the floodlight?

https://github.com/home-assistant/core/blob/0c042e8f7253becf47728a830ca75e32511d9603/homeassistant/components/reolink/light.py#L46-L55

Reply:
See discussion above: <https://github.com/home-assistant/core/pull/87932#pullrequestreview-1323877435>

Reply:
I read up on the comment history in the PR.

Please add a code comment explaining the situation so we don't need to wonder what's going on when reading the code.

Reply:
👍 
CC @starkillerOG

Reply:
I will add a code comment and some extra documentation because this is indeed not clear enough, just a moment

Reply:
Documentation PR is here: https://github.com/home-assistant/home-assistant.io/pull/26469

Reply:
Code comment PR is here: https://github.com/home-assistant/core/pull/89234